File: quiz-platform-backend/src/main/java/com/quizplatform/service/CategoryService.java
package com.quizplatform.service;

import com.quizplatform.dto.request.CategoryRequest;
import com.quizplatform.dto.response.CategoryResponse;

import java.util.List;

public interface CategoryService {
    CategoryResponse createCategory(CategoryRequest request);
    CategoryResponse updateCategory(Long id, CategoryRequest request);
    void deleteCategory(Long id);
    CategoryResponse getCategoryById(Long id);
    List<CategoryResponse> getAllCategories();
}

File: quiz-platform-frontend/src/services/categoryService.ts
import api from '@/utils/api';
import { 
  CategoryResponse, 
  CreateCategoryRequest, 
  ApiResponse 
} from '@/types';

export const categoryService = {
  getAllCategories: async () => {
    const response = await api.get<ApiResponse<CategoryResponse[]>>('/categories');
    return response.data;
  },

  getCategoryById: async (id: number) => {
    const response = await api.get<ApiResponse<CategoryResponse>>(`/categories/${id}`);
    return response.data;
  },

  createCategory: async (data: CreateCategoryRequest) => {
    const response = await api.post<ApiResponse<CategoryResponse>>('/categories', data);
    return response.data;
  },

  updateCategory: async (id: number, data: CreateCategoryRequest) => {
    const response = await api.put<ApiResponse<CategoryResponse>>(`/categories/${id}`, data);
    return response.data;
  },

  deleteCategory: async (id: number) => {
    const response = await api.delete<ApiResponse<void>>(`/categories/${id}`);
    return response.data;
  }
};
